### PR TITLE
Resolve mysql deprecations in tests

### PR DESCRIPTION
**Pull request by dirrao**
Resolve mysql deprecations in tests
related: https://github.com/apache/airflow/issues/38642

### CODE REVIEW

**Review by Lee-W — APPROVED** [review]

File: tests/providers/mysql/operators/test_mysql.py [hunk 1]
```diff
@@ -139,7 +141,7 @@ class MySqlHookForTests(MySqlHook):
 
     dbapi_hook = MySqlHookForTests()
 
-    class SQLExecuteQueryOperatorForTest(MySqlOperator):
+    class SQLExecuteQueryOperatorForTest(SQLExecuteQueryOperator):
         def get_db_hook(self):
             return dbapi_hook
```
Comment (eladkal) [inline tests/providers/mysql/operators/test_mysql.py]: i am not sure if this is the right way to test?
The `SQLExecuteQueryOperatorForTest` feels strange, we should test the actual operator I don't see a reason why we need to have extra class on top of it for testing

I see we do things differently in Postgres
https://github.com/apache/airflow/blob/e8af807cad1eafb52a5e94c1da15cdff2027696c/tests/providers/postgres/operators/test_postgres.py#L140

Comment (dirrao) [reply tests/providers/mysql/operators/test_mysql.py]: Similar test is covered in other test case `test_mysql_operator_openlineage`. One addition in this test cases they are comparing the schema.

Comment (eladkal) [reply tests/providers/mysql/operators/test_mysql.py]: But why do we need the custom class to achieve that?

Comment (dirrao) [reply tests/providers/mysql/operators/test_mysql.py]: Not required. However, the test case is written in that way to inject the mocked dbapi_hook. 
I have updated it without additional class.

**Review by potiuk — APPROVED** [review]

**Pull request merged**